bin/node/cli/src/service.rs

```diff
@@ -99,30 +99,46 @@ macro_rules! new_full_start {
 				import_setup = Some((block_import, grandpa_link, babe_link));
 				Ok(import_queue)
 			})?
-			.with_rpc_extensions(|builder| -> std::result::Result<RpcExtension, _> {
-				let babe_link = import_setup.as_ref().map(|s| &s.2)
-					.expect("BabeLink is present for full services or set up failed; qed.");
+			.with_rpc_extensions(|builder| {
```

Comment:
We could expect a closure that get's a `RpcDependencies` object here instead of `Builder`, which would contain all required stuff and `deny_unsafe`. That would simplify the code a bit (no extra closure), but would make this method special in a way that it's not invoked right away like other `with_*` methods.